### PR TITLE
[WIP] system-monitor: bcc -> libbpf

### PR DESCRIPTION
**Pull request by akshatagarwl**
Some TODOs (or maybe they should come in a different PR?)

- [ ] Remove the bcc dependencies from {Make, Vagrant}files and the likes
- [ ] Use tracepoints for instrumenting common syscalls instead of kprobes

### CODE REVIEW

**Comment by akshatagarwl** [comment]
cc @weirdwiz @oneiro-naut @geyslan for review

**Comment by nam-jaehyun** [comment]
This PR is for #38 and it will be merged when KubeArmor supports libbpf inside the KubeArmor daemonset.

File: KubeArmor/monitor/systemMonitor.go [hunk 1]
```diff
@@ -245,139 +245,133 @@ func (mon *SystemMonitor) InitBPF() error {
...
-	}
-	bpfSource := string(content)
-
 	mon.Logger.Print("Initializing an eBPF program")
 
 	if mon.EnableKubeArmorPolicy && !mon.EnableKubeArmorHostPolicy { // container only
-		mon.BpfModule = bcc.NewModule(bpfSource, []string{"-O2"})
-		if mon.BpfModule == nil {
-			return errors.New("bpf module is nil")
+		mon.BpfModule, err = lbpf.OpenObjectFromFile(bpfObjPath + "system_monitor.bpf.o")
```
Comment (oneiro-naut) [inline KubeArmor/monitor/systemMonitor.go:261]: Does this function call assume that the object file is already there(ie the bpf code is already compiled)?

Comment (akshatagarwl) [reply KubeArmor/monitor/systemMonitor.go:261]: Sorry didn't see this before. Yes it does assume that

**Pull request merged**